File: Data Structures & Algorithms/reorder-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def reorderList(self, head: Optional[ListNode]) -> None:
        slow = head
        fast = head
        while fast and fast.next:
            slow = slow.next
            fast = fast.next.next

        # slow is in middle now
        # reverse second half of the list
        node = slow.next
        prev = None
        slow.next = None
        while node:
            tmp = node.next
            node.next = prev
            prev = node
            node = tmp

        slow = prev
        node = head
        while slow:
            tmp1 = node.next
            tmp2 = slow.next
            node.next = slow
            slow.next = tmp1
            node = tmp1
            slow = tmp2

File: Data Structures & Algorithms/is-anagram/submission-0.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size()!=t.size())
            return false;
        int p[26]={0};
        for(auto &c: s){
            p[c - 'a']++;
        }
        for(auto &c: t){
            p[c - 'a']--;
        }
        for(int i:p){
            if(i)
                return false;
        }
        return true;
    }
};
